helm/cluster/templates/cluster-api/cluster.yaml

```diff
@@ -0,0 +1,36 @@
+apiVersion: cluster.x-k8s.io/v1beta1
+kind: Cluster
+metadata:
+  annotations:
+    {{- with .Values.metadata.description }}
+    cluster.giantswarm.io/description: "{{ . }}"
+    {{- end }}
+  labels:
+    {{- include "labels.common" $ | nindent 4 }}
+    cluster-apps-operator.giantswarm.io/watching: ""
+    {{- if .Values.metadata.labels }}
+    {{- range $key, $val := .Values.metadata.labels }}
+    {{ $key }}: {{ $val | quote }}
+    {{- end }}
+    {{- end }}
+  name: {{ include "resource.default.name" $ }}
+  namespace: {{ .Release.Namespace }}
+spec:
+  clusterNetwork:
+    services:
+      cidrBlocks:
+      {{- toYaml .Values.connectivity.network.services.cidrBlocks | nindent 8 }}
+    pods:
+      cidrBlocks:
+      {{- toYaml .Values.connectivity.network.pods.cidrBlocks | nindent 8 }}
+  controlPlaneRef:
```

Comment:
Inside `controlPlaneRef` and `infrastructureRef` we do have the field `namespace` which we could define already

Reply:
We don't set namespace in these `*Ref` fields, as then it uses the same namespace as the resource which has the reference (`Cluster` here).

I think there is/was even a discussion in upstream CAPI to change the reference type to something like `LocalReference` so you cannot even set the namespace even if you wanted.

Reply:
I see that make sense